torchci/pages/minihud/[repoOwner]/[repoName]/[branch]/[[...page]].tsx

```diff
@@ -231,12 +234,161 @@ function CommitSummaryLine({
           </a>
         </span>
       )}
+      {ttsAlert && (
+        <span style={{ float: "right" }}>
+          <b>tts alert</b>
+        </span>
+      )}
       <CommitLinks row={row} />
     </div>
   );
 }
 
-function CommitSummary({ row }: { row: RowData }) {
+function DurationInfo({
+  jobs,
+  prevRow,
+  expandAllDurationInfo,
+}: {
+  jobs: JobData[];
+  prevRow: RowData | undefined;
+  expandAllDurationInfo: boolean;
+}) {
+  function getAggregateTestTimes(jobs: JobData[] | undefined) {
+    return _.reduce(
+      jobs,
+      (
+        prev: {
+          [key: string]: {
+            duration: number;
+            validData: boolean;
+          };
+        },
+        cur
+      ) => {
+        // filter out rocm and macos because they tend to be variable
+        if (
+          cur.name != undefined &&
+          cur.name.includes(",") &&
+          !cur.name.includes("rocm") &&
+          !cur.name.includes("macos")
+        ) {
+          let name = cur.name.substring(0, cur.name.indexOf(","));
+          if (!(name in prev)) {
+            prev[name] = { duration: 0, validData: true };
+          }
+          if (cur.conclusion != "success" || cur.durationS === undefined) {
+            prev[name].validData = false;
+          } else {
+            prev[name].duration += cur.durationS;
+          }
+        }
+        return prev;
+      },
+      {}
+    );
+  }
+
+  const prevRowJobsAggregate = _.pickBy(
+    getAggregateTestTimes(prevRow?.jobs),
+    (value) => value.validData
+  );
+
+  function getDurationInfo(name: string, duration: number, validData: boolean) {
+    const durationString = validData ? durationHuman(duration) : "invalid data";
```

Comment:
maybe N/A is more general than "invalid data"? As invalid sounds like something is wrong, but this usually just happens when one of the workflows didn't succeed or is still pending.

this is just a semantic preference tho